features/old/stats/provider_side.feature

```diff
@@ -47,7 +47,17 @@ Feature: Provider stats
     And I follow "Top Applications"
     Then I should see "Top Applications" in a header
 
-
+  Scenario: Default metric is product metric
+    Given provider "foo.3scale.localhost" has multiple applications disabled
+    When I log in as provider "foo.3scale.localhost"
+    Given a product
+    And a backend
+    And the backend is used by this product
+    When I go to the overview page of product "API"
+      And I follow "Analytics"
+      And I follow "Traffic"
+    Then I should see "Hits (hits)"
+    Then I should not see "Hits (hits.55)"
```

Comment:
About whether the `I should see` step finds hidden items, I did  a quick research and I think it doesn't. Because:

1. The step uses the method should `have_content`
2. `have_content` is an alias for `have_text` [code](https://github.com/teamcapybara/capybara/blob/master/lib/capybara/rspec/matchers.rb#L133)
3. `have_test` ends up calling `assert_text`
4. We are not sending any option to `assert_text` , and also don't set `ignore_hidden_elements`  (AFAIK). In this context the [docs](https://github.com/teamcapybara/capybara/blob/master/lib/capybara/node/matchers.rb#L654) say only visible elements are considered.

About this cucumber, I still think it doesn't assure the issue is solved. If I place the debugger to see in which order the metrics are being received from the controller , I get:

```
0> @metrics.first
=> #<Metric id: 2, system_name: "hits", description: "Number of API hits", unit: "hit", created_at: "2023-04-05 15:05:51", updated_at: "2023-04-05 15:05:51", service_id: 2, friendly_name: "Hits", parent_id: nil, tenant_id: 2, owner_id: 2, owner_type: "Service">
```
So the first metric is for a service and we cannot be sure if the default metric is a service one by coincidence or because of the code. You have to ensure the first metric received is a backend one, so if the default metric is for a service, it's because the code did it.

Reply:
why do you think that in response it should always  be backend one, suppose you have total of 5 metric and out of which only one is of backend.

so we have added check in the view to filter it by condition  of checking owner_type. And in specs also for a given product and backend by default they have  2 metric on for each owner type and hits and hits.55 are their system name

so if we he Hits (hits)  then definitely its a service one

Reply:
I can check the 4 points you mentioned

Reply:
> so if we he Hits (hits) then definitely its a service one

That's the expected result, but you can't know either it's correct because of your patch or because the service one happens to be the first in the list. If you run this test on master it will pass too, because it's just selecting the first in the item in the list, which happens to be `Hits (hits)`.

Reply:
so what kind of test you are expecting. I am not sure at what extent we can make thing check with cucumber

Reply:
I expect the same test, but receiving a backend metric first in the list

Reply:
I don't know how to do that. Can you give example.